### ISSUE TITLE
Incorporate speckle reduction algorithms?

### ISSUE THREAD

**Issue by csheaff** [KitwareMedical/ITKUltrasound#52]
Hi, I just discovered ITKUltrasound. As you guys probably know, reducing speckle can have a significant impact on image segmentation. Have you thought about incorporating a suite of different algorithms for doing this? Might want to take a look at this MATLAB toolbox and a paper written about it:

https://www.mathworks.com/matlabcentral/fileexchange/54044-image-despeckle-filtering-toolbox?focused=5709994&tab=function

http://www.sciencedirect.com/science/article/pii/S0169260714000327

**Comment by thewtex** [comment]
@csheaff Yes, we intend to add despeckle methods. Thank you for the link and suggestion. It looks like a good open source / open science algorithm. We will add it here, and our implementation should be real-time. I will also add some additional despeckling algorithms. Any other addition suggestions or contributions are welcome.

**Comment by csheaff** [comment]
great. i might also suggest adding Oriented Speckle Reduction Anisotropic Diffusion (OSRAD) by Krissian. From what i've read from review papers, it offers the best balance between performance and efficacy. There's a C++ implementation of his somewhere on the web that is free for download, but I can't seem to find it.

**Comment by thewtex** [comment]
@csheaff Excellent, thanks!

**Comment by agirault** [comment]
As discussed with @tbirdso and @vicory, the OSRAD source code is LGPL and cannot be integrated into ITK as-is, so it might be reimplemented in ITK to be compatible with Apache 2.0.

@karlkrissian we saw you had contributed to ITK/VTK/Slicer in the past which was very exciting to us! Let us know if you'd be interested in helping port the OSRAD algorithm to ITK: even just getting your guidance & reviews would be extremely beneficial.

**Comment by karlkrissian** [comment]
Hello, I will probably be able to help during my free time.

**Comment by agirault** [comment]
@karlkrissian that's amazing, thanks a lot! Is there a world where you (and other contributors to OSRAD) would agree to relicense it from LGPL to Apache 2? That would be mostly for convenience to be able to reuse some of the code in ITK without worrying about copy-pasting limitations.

**Comment by agirault** [comment]
Merci @karlkrissian!

**Comment by karlkrissian** [comment]
It is ok for me to relicense the code from LGPL to Apache 2, before doing
it, I will have to check two files contributed by another author:
CurvaturasPrincipales.*pp.

Le ven. 16 juil. 2021 à 12:13, Alexis Girault ***@***.***> a
écrit :
